multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/MemoryApplicationAttributeUpdater.java

```diff
@@ -1,25 +1,30 @@
 package org.cloudfoundry.multiapps.controller.process.util;
 
+import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudApplicationExtended;
 import org.cloudfoundry.multiapps.controller.process.util.ElementUpdater.UpdateStrategy;
 
 import com.sap.cloudfoundry.client.facade.domain.CloudApplication;
+import com.sap.cloudfoundry.client.facade.domain.CloudProcess;
 
 public class MemoryApplicationAttributeUpdater extends ApplicationAttributeUpdater {
 
-    public MemoryApplicationAttributeUpdater(Context context) {
+    private final CloudProcess existingProcess;
+
+    public MemoryApplicationAttributeUpdater(Context context, CloudProcess process) {
         super(context, UpdateStrategy.REPLACE);
+        this.existingProcess = process;
     }
 
     @Override
-    protected boolean shouldUpdateAttribute(CloudApplication existingApplication, CloudApplication application) {
+    protected boolean shouldUpdateAttribute(CloudApplication existingApplication, CloudApplicationExtended application) {
         if (application.getMemory() == 0) { // 0 means "not specified".
             return false;
         }
-        return existingApplication.getMemory() != application.getMemory();
+        return existingProcess.getMemoryInMb() != application.getMemory();
```

Comment:
Does `application.getMemory()` returns number in megabyte unit?

Reply:
Same as the disk quota. Will check.

Reply:
Yes, the module memory is in megabytes.